job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/WeblogicDeployerContext.groovy

```diff
@@ -0,0 +1,151 @@
+package javaposse.jobdsl.dsl.helpers.publisher
+
+import javaposse.jobdsl.dsl.Context
+import javaposse.jobdsl.dsl.ContextHelper
+import org.apache.commons.lang.RandomStringUtils
+
+/**
+ * <p>DSL Support for the weblogic-deployment-plugin.</p>
+ * <p><a href="https://wiki.jenkins-ci.org/display/JENKINS/WebLogic+Deployer+Plugin">WebLogic Deployer Plugin</a></p>
+ */
+class WeblogicDeployerContext implements Context {
+
+    /**
+     * <p>Enumeration of available deployment stage modes.</p>
+     */
+    static enum WeblogicDeploymentStageModes {
+
+        BY_DEFAULT('bydefault'),
+
+        STAGE('stage'),
+
+        NO_STAGE('nostage'),
+
+        EXTERNAL_STAGE('external_stage')
+
+        private final String stringRepresentation
+
+        WeblogicDeploymentStageModes(String stringRepresentation) {
+            this.stringRepresentation = stringRepresentation
+        }
+
+        @Override
+        String toString() {
+            this.stringRepresentation
+        }
+    }
+
+    List<Node> taskNodes = []
+    List<Node> deploymentPoliciesIdsNodes = []
+
+    boolean mustExitOnFailure = false
+    boolean forceStopOnFirstFailure = false
+    boolean deployingOnlyWhenUpdates = false
+    String deployedProjectsDependencies = ''
+
+    /**
+     * <p>Fails the build if the deployment fails.</p>
+     *
+     * @param mustExitOnFailure true, false (default)
+     */
+    void mustExitOnFailure(boolean mustExitOnFailure = true) {
+        this.mustExitOnFailure = mustExitOnFailure
+    }
+
+    /**
+     * <p>Stop the job on first deployment failure. No other defined
+     * deployment task of this job will be executed.</p>
+     * @param forceStopOnFirstFailure true, false (default)
+     */
+    void forceStopOnFirstFailure(boolean forceStopOnFirstFailure = true) {
+        this.forceStopOnFirstFailure = forceStopOnFirstFailure
+    }
+
+    /**
+     * <p>Deploy only if the build was triggered by a parameterized
+     * cause AND the SCM detects changes.</p>
+     * @param deployingOnlyWhenUpdates true, false (default)
+     */
+    void deployingOnlyWhenUpdates(boolean deployingOnlyWhenUpdates = true) {
+        this.deployingOnlyWhenUpdates = deployingOnlyWhenUpdates
+    }
+
+    /**
+     * <p>(experimental plugin feature) Defines a dependency to other deployment jobs.</p>
+     * @param deployedProjectsDependencies job name of an other deployment job
+     */
+    void deployedProjectsDependencies(String deployedProjectsDependencies) {
+        this.deployedProjectsDependencies = deployedProjectsDependencies
+    }
+
+    void deploymentPolicies(Closure deploymentPoliciesClosure = null) {
```

Comment:
Having a closure and enum values is a duplicate construct. Either use the closure with explicit methods for each option like this:

```groovy
deploymentPolicies {
    legacyCode()
    remoteHost()
}
```

or use a varargs enum argument:

```groovy
deploymentPolicies(LEGACY_CODE, USER_ID)
```